pkg/controller/acmechallenges/sync.go

```diff
@@ -237,7 +237,7 @@ func handleError(ch *cmacme.Challenge, err error) error {
 	if acmeErr, ok = err.(*acmeapi.Error); !ok {
 		ch.Status.State = cmacme.Errored
 		ch.Status.Reason = fmt.Sprintf("unexpected non-ACME API error: %v", err)
-		logf.V(logf.ErrorLevel).ErrorS(err, "unexpected non-ACME API error")
+		logf.FromContext(ctx).V(logf.ErrorLevel).Error(err, "unexpected non-ACME API error")
```

Comment:
I **think** that errors always get logged, regardless of the level...but I've never dug into the code to check whether that's true.
If that is true, then what is the point of having `logf.ErrorLevel` here (or anywhere)?
This is only an observation, not a blocker, because I see that the previous logged error was logged with a level.

Reply:
Indeed, I think `Error` + loglevel does not make sense.
Probably something that can be improved in other places too.